test/unit/lib/plugins/aws/package/lib/generateCoreTemplate.test.js

```diff
@@ -95,6 +95,26 @@ describe('#generateCoreTemplate()', () => {
       });
     }));
 
+  it('should enable S3 bucket versioning if specified', () =>
+    runServerless({
+      config: {
+        service: 'irrelevant',
+        provider: {
+          name: 'aws',
+          deploymentBucket: {
+            versioning: true,
+          },
+        },
+      },
+      command: 'package',
+    }).then(({ cfTemplate }) => {
```

Comment:
Let's use `async/await` instead of `then`

Reply:
Done. Note that all the other tests in this file uses the `then` pattern. Didn't change them all in order to confine the scope of this PR.

> Thanks @mars-lan - great call 👍 I have only minor comments, please let me know what do you think.
> 
> Additionally, it would be good to add a note about this property here: https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/deploying.md#tips

Added.